solr/solrj/src/java/org/apache/solr/client/solrj/impl/HttpJdkSolrClient.java

```diff
@@ -294,15 +294,16 @@ private PreparedRequest preparePutOrPost(
       final PipedInputStream sink = new PipedInputStream(source);
       bodyPublisher = HttpRequest.BodyPublishers.ofInputStream(() -> sink);
 
-      contentWritingFuture =
-          executor.submit(
-              () -> {
-                try (source) {
-                  contentWriter.write(source);
-                } catch (Exception e) {
-                  log.error("Cannot write Content Stream", e);
-                }
-              });
+      executor.submit(
+          () -> {
+            try (source) {
+              contentWriter.write(source);
+              contentWritingFuture.complete(null);
+            } catch (Exception e) {
```

Comment:
Can you change this to `Throwable`? We don't want a possibility of this not being caught.